concent_api/core/tests/utils.py

```diff
@@ -197,13 +202,69 @@ def _test_database_objects(
         task_id                     = None,
         receive_delivered_status    = None,
     ):
-        self.assertEqual(StoredMessage.objects.last().type,           last_object_type.TYPE)
-        self.assertEqual(StoredMessage.objects.last().task_id,        task_id)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().type,      last_object_type.TYPE)
+        self.assertEqual(StoredMessage.objects.order_by('timestamp').last().task_id,   task_id)
 
         if receive_delivered_status is not None:
             self.assertEqual(ReceiveStatus.objects.last().delivered,        receive_delivered_status)
             self.assertEqual(ReceiveStatus.objects.last().message.task_id,  task_id)
 
+    def _test_subtask_state(
+        self,
+        task_id,
+        subtask_id,
+        subtask_state,
+        provider_key,
+        requestor_key,
+        list_of_nested_messages,
+        next_deadline = None,
+    ):
+        self.assertTrue(StoredMessage.objects.filter(subtask_id = subtask_id).exists())
+        subtask = Subtask.objects.get(subtask_id = subtask_id)
+        self.assertEqual(subtask.task_id,              task_id)
+        self.assertEqual(subtask.subtask_id,           subtask_id)
+        self.assertEqual(subtask.state,                subtask_state.name)
+        self.assertEqual(subtask.provider.public_key,  provider_key)
+        self.assertEqual(subtask.requestor.public_key, requestor_key)
+
+        assert Client.objects.filter(public_key = provider_key).exists()
+        assert Client.objects.filter(public_key = requestor_key).exists()
+
+        subtask_deadline = None
+        if subtask.state_enum in Subtask.ACTIVE_STATES:
+            subtask_deadline = subtask.next_deadline.timestamp()
+        self.assertEqual(subtask_deadline, next_deadline)
+
+        self._test_subtask_nested_messages(subtask_id, list_of_nested_messages)
+
+    def _test_subtask_nested_messages(self, subtask_id, list_of_nested_messages):
+        subtask = Subtask.objects.get(subtask_id = subtask_id)
```

Comment:
You can pass subtask directly, there will be one more query less. I know it is just tests but lets keep having good habits. Also, if we will get really lot of tests this can actually speed things up.

Reply:
:+1: